components/server/ee/src/user/account-statement-provider.ts

```diff
@@ -19,20 +20,18 @@ export type CachedAccountStatement = Pick<AccountStatement, "remainingHours" | "
 export class AccountStatementProvider {
     @inject(AccountService) protected readonly accountService: AccountService;
 
-    protected cachedStatement: CachedAccountStatement | undefined;
+    /**
+     * AccountStatements, cached by userId
+     */
+    protected readonly cachedStatements = new GarbageCollectedCache<CachedAccountStatement>(5 * 60, 10 * 60);
 
-    setCachedStatement(cachedStatement: CachedAccountStatement) {
-        this.cachedStatement = cachedStatement;
-    }
-
-    getCachedStatement(): CachedAccountStatement | undefined {
-        return this.cachedStatement;
+    getCachedStatement(userId: string, date: Date): CachedAccountStatement | undefined {
```

Comment:
I still haven't looked very closely into the usage code so please bar with me if this is a stupid question: Why do we pass around the `now` date. Is there another time a user might start a workspace than now resp. can we answer that question?

Reply:
> I still haven't looked very closely into the usage code

FYI: This the is old Chargebee-related code, not usage-based related. (see updated description above :point_up: )

> Why do we pass around the now date?

This mostly is for making sure we're making time-based decision based on the same "reference"/"now". Also, it's helpful for testing.

:cloud: Could be that we do not/no longer need it in all places now - but as this code's lifetime is 6 months max., I don't think it makes sense ~cleanup~ tidy here, now. :+1:

Reply:
:information_source:  This was just a propagation of [this case](https://github.com/gitpod-io/gitpod/pull/12431#discussion_r956186607) which got removed.